### ISSUE TITLE
Support specifying default options through server settings

### ISSUE THREAD

**Issue by rchl** [charliermarsh/ruff-lsp#67]
(Note that I haven't yet played with overriding settings so things I write below might be inaccurate. I did look through documentation though.)

I think it would make sense to be able to tell the server to use certain defaults (like line width 120) but still allow the project-specific overrides win. I believe that if one now specifies default line width through `initializationOptions.settings.args` then that will win over any other settings.

I saw in the documentation that it's possible to specify default settings in a file within home directory and that's fine but I would still prefer to specify defaults through LSP settings instead to keep everything contained within the server configuration.

I wonder what do you think about this.

**Comment by charliermarsh** [comment]
Your description is correct. The thing holding me back is that I'm not sure how best to enable this without further complicating Ruff's settings detection. I guess we could save any defaults to a TOML file locally, and then point Ruff to it (e.g., `--default-config /path/to/tmp.toml`).

**Comment by rchl** [comment]
Sound a bit hacky to generate a random file like that. I guess just supporting `--default-config` and requiring user to generate the file manually might be a better option after all.

**Comment by liebkne** [comment]
Does this mean `ruff-lsp` does not do [config file discovery](https://beta.ruff.rs/docs/configuration/#command-line-interface) like `ruff` by default? I have `ruff.toml` in `~/Library/Application Support/ruff/ruff.toml` but seems to be ignore by `ruff-lsp` but `ruff`.

**Comment by dsully** [comment]
Related - it seems that the new `ruff format` uses the line length from 

```toml
[tool.ruff]
line-length = 120
```

When used on the command line, but there doesn't appear to be any way to pass line-length specifically as format args via the LS. If a config file on disk does not exist, but `lint.args = ...` was set in the LS config, would formatting behave as expected?

**Comment by charliermarsh** [comment]
No, it wouldn't respect the line length from `lint.args` -- that's only passed to the `ruff check` subcommand, the formatter under-the-hood is running `ruff format`, which doesn't support taking the line length on the command line. We could consider adding it!

**Comment by dsully** [comment]
I think that would be extremely useful (and mirror how `black`'s handles CLI arguments). I wasn't sure if it was better to ask here or in the `ruff` project.

**Comment by MichaReiser** [comment]
I think we shouldn't promote the use of `lint.args` or `format.args` and instead add dedicated settings. My motivation for removing the need for the `.args` settings is that we intend to rewrite our LSP in Rust and directly call into a long-running server instead of spawning a CLI on every keystroke. This will give us much better performance and better error handling but it means that simply passing `args` no longer works (because there's no CLI to call).

**Comment by dsully** [comment]
That's fantastic to hear - my request would then be a way to adjust the linter & formatter behavior via the LS settings.
